File: src/main/Python/boj_step10/boj3009.py
import sys
cin = sys.stdin.readline

xList = [0] * 3
yList = [0] * 3
for i in range(3):
    x, y = map(int, cin().split())
    xList[i] = x
    yList[i] = y
x = list(filter(lambda x: xList.count(x) == 1, xList))
y = list(filter(lambda y: yList.count(y) == 1, yList))
print(x.pop(), y.pop())

File: src/main/Cpp/boj_step10/boj3009.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    vector<int> xList(3);
    vector<int> yList(3);

    for (int i = 0; i < 3; ++i) {
        cin >> xList[i] >> yList[i];
    }

    auto find_unique = [](vector<int>& list) -> int {
        for (int i = 0; i < list.size(); ++i) {
            if (count(list.begin(), list.end(), list[i]) == 1) {
                return list[i];
            }
        }
        return -1;
    };

    int x = find_unique(xList);
    int y = find_unique(yList);

    cout << x << " " << y << endl;

    return 0;
}
